application/controllers/admin/surveyadmin.php

```diff
@@ -782,7 +782,8 @@ public function activate($iSurveyID)
                 Survey::model()->resetCache(); // Make sure the saved values will be picked up
             }
 
-            $aResult = activateSurvey($iSurveyID);
+            $aResult = $surveyActivator->activate();
```

Comment:
Is `$aResult` ever checked?

Reply:
I'm not sure about the controller. I was only aiming to replace the helper function with a class doing the same thing.